### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Nileso**

### CODE REVIEW

**Review by sebzera — COMMENTED** [review]
The code functions and meets all of the running requirements of the assignment. The error handling for the die roll parameters and the matching sum values is handled well, and the code does not erroniously generate different values. Commenting could be improved by describing what some methods do, even if it seems obvious. The code is easy to read, is styled properly and the variables are named appropriately.

**Comment by Lemony-Juicy** [comment]
Well, I'll tell ya what, this here code does what it's s'posed to do, yep, it checks all them boxes for what we need it to do. It's got a good grip on keepin' them dice rolls in line and makin' sure them numbers match up just right, without mixin' 'em up all wrong-like. Now, I reckon it could use a bit more jabberin' 'bout what some of them doohickeys (I mean methods) are up to, even if it looks clear as mud. But I gotta say, it's purty darn easy on the eyes, styled up nice and neat, and them names on the variables ain't confusin' one bit.

File: CMP1903_A1_2324/Die.cs [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
+        //Method
+        public int Roll()
+        {
+            currentValue = random.Next(1, 7); //creates a random number for current value
+            return currentValue;
+        }
+
+
+    }
+}
```
Comment (naruto08060119487) [inline CMP1903_A1_2324/Die.cs:28]: Clear consise and successfully generate a random value

File: CMP1903_A1_2324/Game.cs [hunk 2]
```diff
@@ -1,21 +1,42 @@
...
+        }
+        public void OutputDice(int[] nums)//outputs the dice
+        {
+            Console.WriteLine($"First die is {nums[0]}");
+            Console.WriteLine($"Second die is {nums[1]}");
+            Console.WriteLine($"Third die is {nums[2]}");
+            Console.WriteLine($"The total is {nums[0] + nums[1] + nums[2]}");
+        }
+    }
+}
```
Comment (naruto08060119487) [inline CMP1903_A1_2324/Game.cs:42]: the game runs perfectly

File: CMP1903_A1_2324/Program.cs [hunk 3]
```diff
@@ -1,19 +1,26 @@
...
+
+            Game game = new Game();
+            int[] nums = game.ReturnRolls();
+            game.OutputDice(nums);
+
+            /* Create a Testing object to verify the output and operation of the other classes.*/
+            Testing tester = new Testing();
+            tester.Test();
+        }
+    }
```
Comment (naruto08060119487) [inline CMP1903_A1_2324/Program.cs:25]: program class works

File: CMP1903_A1_2324/Testing.cs [hunk 4]
```diff
@@ -1,20 +1,28 @@
...
+            /* Create a Die object and call its method.
+            * Use debug.assert() to make the comparisons and tests.*/
+
+            Die dieTest = new Die();
+            dieTest.Roll();
+            Debug.Assert(dieTest.currentValue >= 1 && dieTest.currentValue <= 6, "Error: Value is not between 1 and 6 inclusive");
+        }
+
+    }
+}
```
Comment (naruto08060119487) [inline CMP1903_A1_2324/Testing.cs:28]: Testing class fire, although would need a bit of inline comments but it compiles and runs perfectly

**Review by naruto08060119487 — APPROVED** [review]

**Pull request merged**